source/UF/H-Levels.lagda

```diff
@@ -202,69 +208,32 @@ From Univalence we can show that (ℍ n) is of level (n + 1), for all n : ℕ.
  where
   C : (X X' : ℍ zero 𝓤) → is-contr (X ＝ X')
   C (X , X-is-contr) (X' , X'-is-contr) =
-    hlevel-closed-under-equiv zero ((X , X-is-contr) ＝ (X' , X'-is-contr))
-                              (X ≃ X') e C'
+    hlevel-closed-under-equiv zero e C'
    where
-    e = ((X , X-is-contr) ＝ (X' , X'-is-contr)) ≃⟨ ≃-sym (to-subtype-＝-≃
-                                                  (λ X → being-singleton-is-prop
-                                                         (fe 𝓤 𝓤))) ⟩
+    I = ≃-sym (to-subtype-＝-≃ (λ X → being-singleton-is-prop fe))
+    e = ((X , X-is-contr) ＝ (X' , X'-is-contr)) ≃⟨ I ⟩
         (X ＝ X')                                ≃⟨ univalence-≃ ua X X' ⟩
         (X ≃ X')                                 ■
     P : is-prop (X ≃ X')
-    P = ≃-is-prop fe (is-prop'-implies-is-prop
-                        (hlevels-are-upper-closed zero X' X'-is-contr))
+    P = ≃-is-prop fe' (is-prop'-implies-is-prop
+                  (hlevels-are-upper-closed zero X' X'-is-contr))
     C' : is-contr (X ≃ X')
     C' = pointed-props-are-singletons (singleton-≃ X-is-contr X'-is-contr) P
 ℍ-is-of-next-hlevel (succ n) 𝓤 ua (X , l) (X' , l') =
```

Comment:
I think you can clean up `ℍ-is-of-next-hlevel` quite a bit by first proving the auxiliary lemma
```
equiv-preserves-hlevel : (n : ℕ) {X : 𝓤 ̇  } {Y : 𝓥 ̇  }
                       → X is-of-hlevel n
                       → Y is-of-hlevel n
                       → (X ≃ Y) is-of-hlevel n
equiv-preserves-hlevel zero = {!!}
equiv-preserves-hlevel (succ n) = {!!}
```
Then you can prove `ℍ-is-of-next-hlevel` without pattern matching on `n`. It also removes the some of the duplication in the correct argument.

Reply:
Note that the zero case follows immediately from `singleton-≃`.

Reply:
Shouldn't the fact that an equivalence between singletons is itsself a singleton already exist elsewhere?

Reply:
To clarify I am fine with inlining it in my lemma but it just seems to be something that one would want developed in the general theory.

Reply:
> Shouldn't the fact that an equivalence between singletons is itself a singleton already exist elsewhere?

Yes, it's `singleton-≃` as I mentioned.

Reply:
That simply gives the center not the contraction.

Reply:
Ah, sorry. Then I'll add it tomorrow (I'm afk atm).

Reply:
I think this is somewhere in TypeTopology. I'll try to find it later.

Reply:
@IanRay11 It's now merged as `≃-is-singleton` in `UF.Singleton-Properties`.

Reply:
Thank you @tomdjong!